### PR TITLE
Changes to system generated emails

### PR DESCRIPTION
**Pull request by briri**
- Updated email messaging per https://github.com/CDL-Dryad/dryad-product-roadmap/issues/222
- Added logic to prevent users who have already registered their ORCID from receiving an invitation
- Moved `submission_success` mailer from Stash:Repo:Repository to CurationActivity
- Added variables for common/default values to ApplicationMailer
- Put the email signature in the mailer layout
- CurationActivity now sends email when submitted, peer_review, published or embargoed and ORCID invitations when published
- added tests and adjusted existing tests

Not sure where this went. it was around at one point:
- added logic back in that sets to peer_review or submitted based on whether or not the identifier has internal_datum

Also:
- Fixed double render bug on downloads controller (found after following sharing link from email)

### CODE REVIEW

File: stash_engine/app/views/stash_engine/user_mailer/peer_review.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<p>Dear @user.name,</p>
```
Comment (sfisher) [inline stash_engine/app/views/stash_engine/user_mailer/peer_review.html.erb]: You probably ant this to be <%= @user.name %> here, right?

Comment (briri) [reply stash_engine/app/views/stash_engine/user_mailer/peer_review.html.erb]: yes. discovered that and some other issues with the email/view rendering this morning.

File: stash_engine/app/views/stash_engine/user_mailer/published.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<p>Dear @user.name,</p>
```
Comment (sfisher) [inline stash_engine/app/views/stash_engine/user_mailer/published.html.erb]: Probably also <%= @user.name %> here instead of @user.name, I think.

Comment (sfisher) [inline stash_engine/app/views/stash_engine/user_mailer/published.html.erb]: probably want to do <%= @user.name %> here.

File: stash_engine/app/views/stash_engine/user_mailer/submitted.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<p>Dear @user.name,</p>
```
Comment (sfisher) [inline stash_engine/app/views/stash_engine/user_mailer/submitted.html.erb]: User name not using the <%= %> here.

**Comment by sfisher** [comment]
Hi Brian,

I noticed a few minor things above and also some test failures in travis.ci introduced by the changes.

Looks pretty good besides if you can look into those things.

**Comment by sfisher** [comment]
Thanks for messing with all this, Brian.  I'm going to merge.

**Pull request merged**